### DISCUSSION TITLE
Issue regarding multiple audio track support on webrtc

### DISCUSSION THREAD

**Discussion by jaggi-mohit** [OvenMediaLabs/OvenMediaEngine#1532] [Help] [answered]
Hi everyone, 
Currently, I am facing an issue regarding multiple audio tracks in webrtc. I tried udp with mulitple audio tracks conversion to webrtc through ovenmediaengine and it converted to webrtc but when playing on ovenplayer there's no option for track selection.
Moreover, how can I verify that my webrtc does contain multiple audio tracks? I have attached a screenshot of logs for your refrence.

Thanks
![Screenshot from 2024-02-21 11-47-32](https://github.com/AirenSoft/OvenMediaEngine/assets/64644750/ea9b63da-11b2-4b97-a7c6-9e9d4bd99158)

**Comment by bchah** [comment]
I don't think OvenPlayer offers audio track selection in the user interface. However you could use the OvenPlayerInstance.getMediaElement() function to get the native media element then look at the underlying [audioTracks](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/audioTracks) for more controls.

To your second question, you can see the two audio tracks in the logs but you can also get this information on the fly using the [OvenMediaEngine API](https://airensoft.gitbook.io/ovenmediaengine/rest-api/v1/virtualhost/application/stream#get-stream-info).

**Reply by jaggi-mohit** [reply]
Hi bchah,
Thanks for the reply mate , I have created an instance of ovenplayer like this -
```
> var player = OvenPlayer.create('player_id', {
>     autoStart: {
>       type: true
>     },
> 
>     
>     waterMark: {
>       image: '/static/img/pmsl.jpg',
>       position: 'top-right',
>       y: '30px',
>       x: '30px',
>       width: '100px',
>       height: '50px',
>       opacity: 1.0
>     },
>     sources: [
> 
>       {
>         label: 'label_for_dash',
>         // Set the type to 'dash'
>         type: 'dash',
>         file: weburl, // specifying my webrtc url
> 
>       }
>     ],
> 
>   });

//and then used  player instance for further operations like this

> var mediainstance= player.getMediaElement()

//it returned a video element  instance and then 
`

> mediainstance.audioTracks

`
// to fetch audio tracks (as guided by you)  but it returned none instead. 

```
Kindly help me out with  the possible scenario for fetching the audio tracks right way

Thanks

**Reply by bchah** [reply]
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/33604674/489e7225-0d0d-45f2-92db-5684e8972192)

Interesting, I get an AudioTrackList. Is what you're showing above the complete code or is there a chance that the 'player' var has fallen out of scope by the time you are calling getMediaElement on it?

As an aside, I notice you're initializing with type 'dash' yet indicating a webrtc url, so the type should also be 'webrtc'.

**Reply by jaggi-mohit** [reply]
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/64644750/24541ccc-9706-4012-92c2-8510047e6ee6)
The player hasn't fallen out of scope it is showing a video element when a player.getMediaElement is called but when I try to fetch audioTracks it shows undefined.

Btw how did you initialize your player? and on which browser you are testing

**Reply by bchah** [reply]
I initialized using the same player.create() as you - but the audioTracks feature seems limited to Safari and Edge only at the moment. I wonder if the only way to do this with OME is via HLS playback and the hls.js audioTrack functions. You may want to take this discussion over to the OvenPlayer forum and see if @SangwonOh has any ideas 😄

**Reply by jaggi-mohit** [reply]
ok no problem 😃 btw thanks for the help mate !!..

**Comment by getroot** [comment]
What does multi audio track mean?
Did you input multiple audio tracks through SRT? Then it can only be selected in LLHLS. WebRTC does not yet support multi audio track playback.
Or do you mean ABR? In that case, you can use the playlist.

**Reply by jaggi-mohit** [reply]
Hi @getroot 
Yes I have udp stream with multiple audio tracks which I have provided to ovenmediaengine in order to convert into webrtc output. Now I want to select those audio tracks via webrtc player. Is it feasible?
By multiple audio track i mean different audios not ABR.

**Reply by getroot** [reply, marked as answer]
You are sending a stream over MPEG-2 TS/UDP. I was a little confused about the UDP stream.

As mentioned before, multi audio track playback is supported only in LLHLS. OME's WebRTC does not yet support multiple audio tracks.

**Reply by jaggi-mohit** [reply]
Okay, @getroot  Thanks for the answer !!
